src/Contexts/Authentication/AuthenticationContext.tsx

```diff
@@ -0,0 +1,49 @@
+import React, {useState, useContext, createContext, ReactNode} from "react";
+import SubmarineAuthenticationHttpClient from "../../SubmarineHttp/Resources/Authentication/SubmarineAuthenticationHttpClient";
+import SubmarineHttpVersion from "../../SubmarineHttp/SubmarineHttpVersion";
+import ISubmarineRegisterRequest from "../../SubmarineHttp/Resources/Authentication/Register/ISubmarineRegisterRequest";
+import ISubmarineAuthenticateRequest from "../../SubmarineHttp/Resources/Authentication/Authenticate/ISubmarineAuthenticateRequest";
+import ISubmarineAuthenticatedResponse from "../../SubmarineHttp/Resources/Authentication/Authenticate/ISubmarineAuthenticatedResponse";
+import AuthenticationIdentity from "./AuthenticationIdentity";
+import IAuthenticationContextProviderProps from "./IAuthenticationContextProviderProps";
+import IAuthenticationContext from "./IAuthenticationContext";
+
+const InitialAuthenticationContext = { register: () => {}, authenticate: () => { } } as IAuthenticationContext;
+const AuthenticationContext = createContext(InitialAuthenticationContext)
+
+export const AuthenticationContextProvider = ({ children }: IAuthenticationContextProviderProps) => {
```

Comment:
Convert this to a class, otherwise it'll be difficult to test.

Reply:
Actually, I'm not sure about this, because you'll be asserting the HTTP client methods are called.